flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/com/ververica/cdc/connectors/values/sink/ValuesDataSink.java

```diff
@@ -109,48 +89,47 @@ private static class ValuesSinkWriter extends PrintSinkOutputWriter<Event>
          */
         private final Map<TableId, Schema> schemaMaps;
 
+        private final Map<TableId, List<RecordData.FieldGetter>> fieldGetterMaps;
+
         public ValuesSinkWriter(boolean materializedInMemory) {
             super();
             this.materializedInMemory = materializedInMemory;
             schemaMaps = new HashMap<>();
-        }
-
-        private void initializeState(Collection<TableSchemaState> bucketStates) {
-            bucketStates.forEach(
-                    TableSchemaState ->
-                            schemaMaps.put(
-                                    TableSchemaState.getTableId(), TableSchemaState.getSchema()));
+            fieldGetterMaps = new HashMap<>();
         }
 
         @Override
-        public void write(Event event) {
-            super.write(event);
+        public void write(Event event, Context context) {
             if (event instanceof SchemaChangeEvent) {
                 SchemaChangeEvent schemaChangeEvent = (SchemaChangeEvent) event;
                 TableId tableId = schemaChangeEvent.tableId();
                 if (event instanceof CreateTableEvent) {
-                    schemaMaps.put(tableId, ((CreateTableEvent) event).getSchema());
+                    Schema schema = ((CreateTableEvent) event).getSchema();
+                    schemaMaps.put(tableId, schema);
+                    fieldGetterMaps.put(tableId, SchemaUtils.createFieldGetters(schema));
                 } else {
                     if (!schemaMaps.containsKey(tableId)) {
                         throw new RuntimeException("schema of " + tableId + " is not existed.");
                     }
-                    schemaMaps.put(
-                            tableId,
+                    Schema schema =
                             SchemaUtils.applySchemaChangeEvent(
-                                    schemaMaps.get(tableId), schemaChangeEvent));
+                                    schemaMaps.get(tableId), schemaChangeEvent);
+                    schemaMaps.put(tableId, schema);
+                    fieldGetterMaps.put(tableId, SchemaUtils.createFieldGetters(schema));
                 }
             } else if (materializedInMemory && event instanceof DataChangeEvent) {
                 ValuesDatabase.applyDataChangeEvent((DataChangeEvent) event);
             }
+            // print the detail message to console for verification.
```

Comment:
Should we add a config to control print or not ?

Reply:
address it.